test/rubygems/test_gem.rb

```diff
@@ -302,6 +302,21 @@ def test_activate_bin_path_resolves_eagerly
     assert_equal %w(a-1 b-2 c-1), loaded_spec_names
   end
 
+  def test_activate_bin_path_in_debug_mode
+    a1 = util_spec 'a', '1' do |s|
+      s.executables = ['exec']
+    end
+
+    install_specs a1
+
+    output, status = Open3.capture2e(
+      { "GEM_HOME" => Gem.paths.home, "DEBUG_RESOLVER" => "1" },
+      Gem.ruby, "-I", File.expand_path("../../lib", __dir__), "-e", "Gem.activate_bin_path('a', 'exec', '>= 0')"
```

Comment:
Thank you!

jruby is complaining about a syntax error, no sure why: https://github.com/rubygems/rubygems/pull/3171/checks?check_run_id=509593004#step:4:128

We can workaround it like this

```suggestion
      Gem.ruby, "-I", File.expand_path("../../lib", __dir__), "-e", Shellwords.shellescape("Gem.activate_bin_path('a', 'exec', '>= 0')")
```

Reply:
@enebo Would you be interested in this "syntax error" on JRuby described in the line above? Ignore if not, and have a good day!

Reply:
(**Update**: Updated the PR.)